src/shared/bus-polkit.c

```diff
@@ -556,6 +556,12 @@ int bus_verify_polkit_async_full(
                         log_debug("Found matching previous polkit authentication for '%s'.", action);
                         return r;
                 }
+
+                /* Processing of a PolicyKit checks is canceled on the first auth. error. */
+                assert(!q->denied_action);
+                assert(!q->absent_action);
+                assert(!q->error_action);
+                assert(!sd_bus_error_is_set(&q->error));
```

Comment:
this doesn't appear right to me. i think it would be better if we clearly refused another pk auth attempt here if there's already a failed action. i.e. return -EBUSY here or so if any of these fields are set.

Reply:
I thought about that, but decided to start with a change that preserves current behavior (as I wrote, such situation already causes an assert, just later when processing polkit's response). Since there's a clear preference for returning an error, I'll do that.

Reply:
Done. I've also moved the check to `async_polkit_query_check_action()` to avoid duplication. It felt like a wrong place for asserts, but an additional check seems okay to me.